storage/innobase/row/row0purge.cc

```diff
@@ -943,10 +944,23 @@ static trx_id_t row_purge_remove_sec_if_poss_leaf(purge_node_t *node,
 			}
 
 			if (btr_cur_optimistic_delete(btr_cur, 0, &mtr)
-                            == DB_FAIL) {
+			    == DB_FAIL) {
 				page_max_trx_id = page_get_max_trx_id(
 					btr_cur_get_page(btr_cur));
-                        }
+				if (trx_sys.find(nullptr, page_max_trx_id,
+						 false)) {
+					/* Because this transaction is
```

Comment:
This is not enough. We would require to check also for other active transactions having ID smaller than page_max_trx_id.

Reply:
Good point. Such transactions would not update `PAGE_MAX_TRX_ID`.